src/adagrams.js

```diff
@@ -1,15 +1,106 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const ENGLISH_LETTERS = generateAlphabet();
+  const LETTERS_FREQUENCY = [
+    9, 2, 2, 4, 12, 2, 3, 2, 9, 1, 1, 4, 2, 6, 8, 2, 1, 6, 4, 6, 4, 2, 2, 1, 2,
+    1,
+  ];
+  const NUM_OF_DRAW_LETTERS = 10;
+
+  const lettersInHand = [];
+  let randomIndex;
+  while (lettersInHand.length < NUM_OF_DRAW_LETTERS) {
+    randomIndex = Math.floor(Math.random() * ENGLISH_LETTERS.length);
+
+    if (LETTERS_FREQUENCY[randomIndex]) {
+      lettersInHand.push(ENGLISH_LETTERS[randomIndex]);
+      LETTERS_FREQUENCY[randomIndex] -= 1;
+    } else continue;
+  }
+  // console.log(`lettersInHand=   ${lettersInHand}`);
+  return lettersInHand;
 };
 
+function generateAlphabet(capital = true) {
+  return [...Array(26)].map((_, i) =>
+    String.fromCharCode(i + (capital ? 65 : 97))
+  );
+}
+
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const mapLetters = {};
+  for (const letter of lettersInHand) {
```

Comment:
Nice job using const for declaring letter since you don't reassign letter in the loop